src/content/tools/hooks.md

```diff
@@ -0,0 +1,277 @@
+
+## Use a hook {: #use-hooks-assets }
+
+To add assets to your project, use a hook. For details,
+see the following sections.
+
+### Add dependencies {: #add-dependencies-hooks-assets }
+
+Open your package in a code editor, navigate to its
+`pubspec.yaml`, and then add the following dependencies as
+needed under `dependencies`:
+
+* `hooks`
+* `code_assets`
+
+You will also likely need to add a tool for your specific
+task, such as `native_toolchain_c` for compiling C code.
+
+:::note
+You need to add the dependencies under `dependencies`, not
+`dev_dependencies`. The hooks will be run by packages and
+applications depending on your package, so the
+Dart code needs to be part of the resolution in those
+packages.
+:::
+
+Example dependencies for a build hook:
+
+```yaml title="pubspec.yaml"
+name: native_add_library
+description: Sums two numbers with native code.
+version: 0.1.0
+
+environment:
+  sdk: '>=3.9.0 <4.0.0'
+
+dependencies:
+  ... 
+  code_assets: any
+  hooks: any
+  native_toolchain_c: any
+
+dev_dependencies:
+  ...
+  ffigen: ^18.0.0
+ ```
+
+### Create a build hook script {: #create-hook }
+
+If you want to use a build hook, first create the native
+library that you want to use, and then follow these steps to
+connect a build hook to that library in a Dart script: 
+
+1.  In your Dart project, create or open `hooks/build.dart`. 
+
+1.  In the main method, add a `build` function from
+    `package:hooks/hooks.dart` and link to the native
+    library. For example:
+
+    ```dart title="hooks/build.dart" highlightLines=6
+    import 'package:hooks/hooks.dart';
+    import 'package:logging/logging.dart';
+    import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+    void main(List<String> args) async {
+      await build(args, (input, output) async {
+        final packageName = input.packageName;
+        final cbuilder = CBuilder.library(
+          name: packageName,
+          assetName: '$packageName.dart',
+          sources: ['src/$packageName.c'],
+        );
+        await cbuilder.run(
+          input: input,
+          output: output,
+          logger: Logger('')
+            ..level = Level.ALL
+            ..onRecord.listen((record) => print(record.message)),
+        );
+      });
+    }
+    ```
+
+    * `input`: The input for `hook/build.dart`. This
+      includes the target OS, target architecture,
+      output directory, etc. For details, see
+      the [`BuildInput`][] class.
+
+    * `output`: The output for `hook/build.dart`. This
+      includes the assets to be bundled, and dependencies.
+      For details, see the [`BuildOutputBuilder`][] class.
+
+[`BuildInput`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput-class.html
+[`BuildOutputBuilder`]: {{site.pub-api}}/hooks/latest/hooks/BuildOutputBuilder-class.html
+
+### Generate or download assets {: #generate-assets }
+
+Assets are generated and/or downloaded when you invoke your
```

Comment:
Since you never really invoke a hook directly, maybe better: Assets are generated and/or downloaded according to your hook script when your hook is invoked by the Dart SDK, which autoamtically happens when you use the `run`, `build`, or `test` command on your package.